pkg/operator/hive/hive.go

```diff
@@ -353,7 +375,9 @@ func (r *ReconcileHiveConfig) deployHive(hLog log.FieldLogger, h resource.Helper
 	// Apply shared pod config passed through from the operator deployment
 	hiveDeployment.Spec.Template.Spec.NodeSelector = r.sharedPodConfig.NodeSelector
 	hiveDeployment.Spec.Template.Spec.Tolerations = r.sharedPodConfig.Tolerations
-	hiveDeployment.Spec.Template.Spec.ImagePullSecrets = r.sharedPodConfig.ImagePullSecrets
+	if hiveConfigHasValidImagePullSecretReference(instance) {
+		hiveDeployment.Spec.Template.Spec.ImagePullSecrets = append(hiveDeployment.Spec.Template.Spec.ImagePullSecrets, *instance.Spec.HiveImagePullSecretRef)
+	}
```

Comment:
Just a thought, if you made the func return the ref instead of just checking for it, you could do something like:
```suggestion
	if ref := getImagePullSecretReference(instance); ref != nil {
		hiveDeployment.Spec.Template.Spec.ImagePullSecrets = append(hiveDeployment.Spec.Template.Spec.ImagePullSecrets, ref)
	}
```